### ISSUE TITLE
Show output also for failed tasks?

### ISSUE THREAD

**Issue by demobox** [msoap/shell2http#3]
Using shell2http 1.4 (a fantastic tool - thanks a lot for creating this!) we are observing the following behaviour: that the response for tasks that return a non-zero exit code is simply something like:

```
exec error: exit status 2
```

In order to actually see the error message and yet still know that the task has actually failed, we are resorting to something along the lines of:

```
/my/task '(do_stuff; echo Exit code: $?; exit 0)'
```

Would this be a good candidate for a command-line flag? Or is there an easier way to achieve the desired result (i.e. having `stdout` and/or `stderr` returned even for non-zero exit codes) without the hackery in the example above?

**Comment by msoap** [comment]
Yes, it makes sense to add the option on the command line like `-output-errors` (or `-out-on-fail`...).
And may be another option `-output-stderr`, for return stderr-out of script (now stderr is printed to the console).

**Comment by demobox** [comment]
> And may be another option -output-stderr, for return stderr-out of script (now stderr is printed to the console).

Ah, good idea! So two orthogonal options?

```
-show-errors: returns the standard output even if the command exits with a non-zero exit code
-include-stderr: also returns output written to stderr (default is stdout only)
```

Or would `-show-errors` automatically imply `-include-stderr`? I assume the users would want both most of the time since the error message would likely go to `stderr`. But I guess there are also enough cases where that is not the case that one might want to make the options separate?

I can also imagine cases where you **just** want `-include-stderr`, such as for commands that print diagnostic info to stderr even when they do not fail

**Comment by msoap** [comment]
Yes, two orthogonal options.
Thanks for naming for options:)

**Comment by msoap** [comment]
Implemented in #4, #5
